### PR TITLE
core: Strip BOM in URLLoader text data

### PR DESCRIPTION
**Pull request by Mesteery**

### CODE REVIEW

File: core/src/loader.rs [hunk 1]
```diff
@@ -1576,6 +1576,11 @@ impl<'gc> Loader<'gc> {
                             tracing::warn!("Invalid URLLoaderDataFormat: {}", data_format);
                         }
 
+                        // strips off an optional UTF8 BOM at the beginning
```
Comment (Lord-McSweeney) [inline core/src/loader.rs:1579]: `bytearray::to_string` covers UTF-16 BOMs, and we'd probably like to cover them too. Additionally, it seems that this also applies to url variables

**Comment by Aaron1011** [comment]
Are you sure that this is done in URLLoader, and not in ByteArray? There's already some BOM-related logic in ByteArray, so I wonder if it just needs to be extended

**Comment by Mesteery** [comment]
> Are you sure that this is done in URLLoader, and not in ByteArray? There's already some BOM-related logic in ByteArray, so I wonder if it just needs to be extended

According to uqers, in FP, URLLoader uses URLStream, and so it's done in URLStream/readUTFBytes like in ByteArray

File: core/src/loader.rs (hunk 1 above)
Comment (Mesteery) [reply core/src/loader.rs]: done

File: core/src/loader.rs [hunk 2]
```diff
@@ -1540,6 +1541,40 @@ impl<'gc> Loader<'gc> {
 
                 let mut activation = Avm2Activation::from_nothing(uc);
 
+                fn strip_bom<'a, 'gc: 'a>(
+                    activation: &mut Avm2Activation<'a, 'gc>,
```
Comment (Lord-McSweeney) [inline core/src/loader.rs:1545]: We should probably avoid duplicating this function across here and `byte_array`- could you define it in `byte_array.rs`?

Comment (Mesteery) [reply core/src/loader.rs]: sure

**Comment by Dinnerbone** [comment]
I would really appreciate a test for this!

**Review by Lord-McSweeney — APPROVED** [review]

**Pull request merged**